### PR TITLE
Revise model fields and asset file check/download logic

### LINKED ISSUE #1: Improve test coverage
```
Name                                   Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------------------------
src/wagtail_bynder/__init__.py            15      5      4      1    58%   20, 24-29
src/wagtail_bynder/blocks.py              15     15      4      0     0%   1-22
src/wagtail_bynder/middleware.py          20      0      8      1    96%   13->31
src/wagtail_bynder/models.py             142     64     20      0    51%   106, 120-129, 132, 135-136, 149-152, 155-156, 183-188, 202-222, 228-239, 255, 318, 322-324, 328-330, 336-366
src/wagtail_bynder/utils.py               44     23      0      0    48%   26-31, 35-44, 48-57, 74-89
src/wagtail_bynder/views/document.py      44      3      4      1    92%   14, 28, 31
src/wagtail_bynder/views/image.py         43      3      4      1    91%   14, 28, 31
src/wagtail_bynder/views/mixins.py        37     12     10      1    60%   11, 34-36, 39-44, 47-52
src/wagtail_bynder/views/video.py         47     12      6      1    75%   15, 30, 35-41, 54-56, 67-68
src/wagtail_bynder/wagtail_hooks.py       29      4     18      3    81%   17->exit, 42->exit, 53-54, 68->exit, 71-72
src/wagtail_bynder/widgets.py             30     30      4      0     0%   1-52
----------------------------------------------------------------------------------
TOTAL                                    479    171     86      9    62%
```

- #10
- #12

### LINKED ISSUE #6: 'original' value missing from asset representations for assets marked as 'Private'
This appears to be dependent on permissions applied to the account that the **Permanent Token** used by the server (`BYNDER_API_TOKEN`) - specifically, where they are less permissive than those of the user attached to the **Permanent Token** used by the chooser (`BYNDER_COMPACT_VIEW_API_TOKEN`).

This could be resolved by including more in-depth instructions in the README on creating users/permissions/tokens in Bynder - where we can be explicit about the permissions that the `BYNDER_API_TOKEN` users should have.

### LINKED ISSUE #7: Stop storing asset data that isn't explicitly used
Currently, a slightly reduced version of the asset data is stored in the `metadata` field (with directly mapped field values removed), and there are a few methods and attributes involved in setting the value.

This field was originally added for a couple of reasons:
- To help debug unexpected issues with variations in asset data
- To help populate new custom field values from local data instead of having to reimport data from Bynder somehow (this will be made easier by #18)

The first is now better handled by explicit exceptions, where the full asset data should be available in the stack trace.

The second issue only really exists because there is no easy way to trigger a re-import of all items when making this sort of change, which should soon be resolved by #18!

### PR DESCRIPTION
**Pull request by ababic**
Resolves #1, #7, #8 and #9

Partially resolves #6 (private documents cannot be supported, as the 'original' value is required) 

## Summary of changes:

1. Added `null=True` to `bynder_id`
2. Removed unused `bynder_id_hash` field
3. Removed `metadata` field and the attributes/methods involved in setting it.
4. Added the `original_filesize` field for all asset models, and `original_width` and `original_height` fields for image and video. `asset_file_has_changed()` methods have also been updated to utilise these field values accordingly.
5. Replaced the `bynder_original_filename` field with `source_filename`, which will store the 'original' filename for documents, the 'source derivative' filename for images, and 'primary derivative' filename for videos. This allows private images and videos to be used in Wagtail, and because derivative names are repeated in derivative filenames by default, it should mean that `asset_file_has_changed()` returns `True` when `BYNDER_IMAGE_SOURCE_THUMBNAIL_NAME` is modified.
6. Adds unit tests covering 97% of `wagtail_bynder/models.py`

## Breaking changes

- Users of the app will need to run `python manage.py makemigrations` to reflect abstract model changes in their concrete models, followed by `python manage.py migrate` to apply those changes.
- It is also recommended that the commands added in #18 be used to set new field values in all environments.
-  The `download_asset_file()` method has been removed from all base models. The newly added `update_file()` method is now responsible for downloading files and updating any relevant field values

### CODE REVIEW

**Comment by ababic** [comment]
pre-commit.ci autofix

File: src/wagtail_bynder/models.py [hunk 1]
```diff
@@ -333,8 +375,6 @@ def update_from_asset_data(
         self,
         asset_data: dict[str, Any],
     ) -> None:
-        super().update_from_asset_data(asset_data)
-
         primary_derivative_name = getattr(
             settings, "BYNDER_VIDEO_PRIMARY_DERIVATIVE_NAME", "Web-Primary"
```
Comment (zerolab) [inline src/wagtail_bynder/models.py:379]:
```suggestion
            settings, "BYNDER_VIDEO_PRIMARY_DERIVATIVE_NAME", "WebPrimary"
```

to match the README and the tests settings

File: tests/testapp/migrations/0003_alter_customdocument_options_alter_video_options.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Generated by Django 5.0.4 on 2024-04-05 16:08
```
Comment (zerolab) [inline tests/testapp/migrations/0003_alter_customdocument_options_alter_video_options.py:1]: nitpick (or personal preference) - renegerate the tests in `0001_initial`. While minor, running tests will be a tad speedier

**Review by zerolab — CHANGES_REQUESTED** [review]
left a couple of minor points + we will want to update the changelog with a note on what changed in case someone's building off this already

**Comment by ababic** [comment]
pre-commit.ci autofix

File: tests/testapp/migrations/0003_alter_customdocument_options_alter_video_options.py (hunk 2 above)
Comment (ababic) [reply tests/testapp/migrations/0003_alter_customdocument_options_alter_video_options.py]: No problem. Updated in [2ded683](https://github.com/torchbox/wagtail-bynder/pull/19/commits/2ded683fd9a029de7a9eb0f573dfa3595b493c1a)

**Review by zerolab — APPROVED** [review]
Excellent work here, Andy! 🚀

**Pull request merged**